Lib/test/test_dataclasses.py

```diff
@@ -68,6 +68,20 @@ def test_field_repr(self):
 
         self.assertEqual(repr_output, expected_output)
 
+    def test_field_recursive_repr(self):
+        rec_field = field()
+        rec_field.type = rec_field
+        rec_field.name = "id"
+        repr_output = repr(rec_field)
+        expected_output = "Field(name='id',type=...," \
```

Comment:
I think I'd just test that "type=..." is in the repr, not for the rest of the values. If new ones are added, this test would need to be modified.

Also, I think an additional test for the original problem:
```python
@dataclass
class D:
    C: C = field()
```
would be good.

Otherwise looks good!

Reply:
And to be clear: I'd just add the second test to make sure it doesn't raise `RecursionError`, I wouldn't go nuts on testing the exact output.

Reply:
Let me know if the assertion in the new test qualifies as "going nuts" -- just seemed easy to check, and more confusing to have a test with no assertions at all.

Reply:
Maybe "go nuts" wasn't the best wording to use :smiley:. Your change is exactly what I would have done, down to the commas, so I think it's perfect. Thanks!